packages/gatsby/src/redux/reducers/pages.ts

```diff
@@ -1,31 +1,44 @@
-const normalize = require(`normalize-path`)
+import normalize from "normalize-path"
+import { ActionsUnion, IGatsbyPage, IGatsbyState } from "../types"
 
-module.exports = (state = new Map(), action) => {
+export const pagesReducer = (
+  state: IGatsbyState["pages"] = new Map<string, IGatsbyPage>(),
+  action: ActionsUnion
+): IGatsbyState["pages"] => {
   switch (action.type) {
     case `DELETE_CACHE`:
       return new Map()
+
     case `CREATE_PAGE`: {
       action.payload.component = normalize(action.payload.component)
-      if (!action.plugin && !action.plugin.name) {
+
+      // throws an error if the page is not created by a plugin
+      if (!action.plugin || !action.plugin?.name) {
```

Comment:
```suggestion
      if (!action.plugin?.name) {
```
This is redundant with the optional chain

Reply:
Good catch @ascorbic! I amended my comment with this change :+1: